### PR TITLE
(#794) New job opportunity should reflect on website also

### PR DESCRIPTION
**Pull request by atul5238**
Target #794 
## Summary

1. New feature integrated using Corcel .
2. Job opportunities will update/delete/create on ColoredCow website also if updated from Portal.

### CODE REVIEW

File: Modules/HR/Http/Controllers/Recruitment/RecruitmentOpportunityController.php [hunk 1]
```diff
@@ -33,4 +35,32 @@ public function index()
...
+            'posted_by' => $validated['by'] ?? null,
+            'link' => $validated['link'] ?? null,
+        ]);
+        $route = $opportunity->type == 'volunteer' ? route('volunteer.opportunities.edit', $opportunity->id) : route('recruitment.opportunities.edit', $opportunity->id);
+        return redirect($route)->with('status', "Successfully updated $opportunity->title!");
+    }
+
+    public function destroy(Job $opportunity)
+    {
+        Corcel::where(['post_type' => 'career', 'post_title' => $opportunity['title']], 0)->delete();
```
Comment (rathorevaibhav) [inline Modules/HR/Http/Controllers/Recruitment/RecruitmentOpportunityController.php:61]: What does `0` do here?

File: Modules/HR/Http/Controllers/Recruitment/RecruitmentOpportunityController.php [hunk 2]
```diff
@@ -33,4 +35,32 @@ public function index()
             'type' => 'recruitment',
         ]);
     }
+
+    public function store(JobRequest $request)
```
Comment (rathorevaibhav) [inline Modules/HR/Http/Controllers/Recruitment/RecruitmentOpportunityController.php:39]: what happens when you update the job from the portal? Does the website counterpart also gets updated?

**Review by rathorevaibhav — CHANGES_REQUESTED** [review]

File: Modules/HR/Http/Controllers/Recruitment/RecruitmentOpportunityController.php [hunk 3]
```diff
@@ -33,4 +35,32 @@ public function index()
...
+    {
+        $validated = $request->validated();
+
+        Corcel::updateOrCreate(
+            ['post_type' => 'career', 'post_title' => $validated['title']],
+            ['post_content' => $validated['description']]
+        );
```
[the remark this answers is not in the document]
Comment (atul5238) [reply Modules/HR/Http/Controllers/Recruitment/RecruitmentOpportunityController.php:46]: This feature includes both update and create, `updateOrcreate` is for just code optimization purpose.

File: Modules/HR/Http/Controllers/Recruitment/RecruitmentOpportunityController.php [hunk 4]
```diff
@@ -33,4 +35,32 @@ public function index()
...
+
+        $opportunity = Job::create([
+            'title' => $validated['title'],
+            'description' => $validated['description'] ?? null, // null needed for backward compatibility
```
[the remark this answers is not in the document]
Comment (atul5238) [reply Modules/HR/Http/Controllers/Recruitment/RecruitmentOpportunityController.php:50]: Yeah I just checkout on UAT, removing description from nullable

File: Modules/HR/Http/Controllers/Recruitment/RecruitmentOpportunityController.php (hunk 1 above)
Comment (atul5238) [reply Modules/HR/Http/Controllers/Recruitment/RecruitmentOpportunityController.php:61]: sorry forgot to remove, removing this.

File: Modules/HR/Http/Controllers/Recruitment/RecruitmentOpportunityController.php (hunk 2 above)
Comment (atul5238) [reply Modules/HR/Http/Controllers/Recruitment/RecruitmentOpportunityController.php:39]: Yes

File: Modules/HR/Http/Controllers/Recruitment/RecruitmentOpportunityController.php [hunk 5]
```diff
@@ -33,4 +35,32 @@ public function index()
...
+            'type' => $validated['type'],
+            'posted_by' => $validated['by'] ?? null,
+            'link' => $validated['link'] ?? null,
```
[the remark this answers is not in the document]
Comment (atul5238) [reply Modules/HR/Http/Controllers/Recruitment/RecruitmentOpportunityController.php:53]: permalink is still generated on Website.

**Comment by rathorevaibhav** [comment]
@am-Anmol just wanted to check if you're working on this branch or on a separate one? If it's a separate one, can we close this Pull Request?

**Comment by am-Anmol** [comment]
@rathorevaibhav Yes, I'm working on a new branch, so closing this pull request.

**Comment by rathorevaibhav** [comment]
Awesome thanks!

**Pull request closed without merging**